codejail/jail_code.py

```diff
@@ -209,11 +210,9 @@ def jail_code(command, code=None, files=None, extra_files=None, argv=None,
         user = COMMANDS[command]['user']
         if user:
             # Run as the specified user
-            cmd.extend(['sudo', '-u', user])
+            cmd.extend(['sudo', '-u', user, 'TMPDIR=tmp'])
```

Comment:
I reverted auto-generating `'TMPDIR=tmp'` from the `env` variable, because to do it right, I had to handle proper shell escaping, (and add a dependency on shellescape to do so, and also check that the variable name was acceptable (no spaces, etc), and then test all of that behavior.  None of which is really necessary to ensure that 'TMPDIR=tmp' is valid.  Since we don't allow users to insert their own values here, I just backed it all out, and went with the simple version.